### PR TITLE
A raw fork/exec driver that provides no isolation.

### PR DESCRIPTION
**Pull request by dadgar**
A raw fork/exec driver that provides no isolation.

Addresses #213

### CODE REVIEW

**Comment by jippi** [comment]
:+1:

File: client/driver/raw_exec.go [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+
+// NewRawExecDriver is used to create a new raw exec driver
+func NewRawExecDriver(ctx *DriverContext) Driver {
+	return &RawExecDriver{*ctx}
+}
+
+func (d *RawExecDriver) Fingerprint(cfg *config.Config, node *structs.Node) (bool, error) {
+	// Check that the user has explicitly enabled this executor.
+	enabled := strings.ToLower(cfg.ReadDefault(rawExecConfigOption, "false"))
+	if enabled == "1" || enabled == "true" {
```
Comment (cbednarski) [inline client/driver/raw_exec.go]: This should use `strconv.ParseBool`.

File: client/driver/raw_exec.go [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+		return nil, fmt.Errorf("missing command for raw_exec driver")
+	}
+
+	// Get the tasks local directory.
+	taskName := d.DriverContext.taskName
+	taskDir, ok := ctx.AllocDir.TaskDirs[taskName]
+	if !ok {
+		return nil, fmt.Errorf("Could not find task directory for task: %v", d.DriverContext.taskName)
+	}
+	taskLocal := filepath.Join(taskDir, allocdir.TaskLocal)
```
Comment (cbednarski) [inline client/driver/raw_exec.go]: Do we verify that this directory exists before writing files to it?

File: website/source/docs/drivers/raw_exec.html.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+description: |-
+  The Raw Exec task driver simply fork/execs and provides no isolation.
+---
+
+# Raw Fork/Exec Driver
+
+Name: `raw_exec`
+
+The `raw_exec` driver is used to execute a command for a task without any
+resource isolation. As such, it should be used with extreme care and is disabled
```
Comment (cbednarski) [inline website/source/docs/drivers/raw_exec.html.md]: s/resource isolation/isolation or resource constraints/ -- I think this is more consistent with phrasing we use elsewhere.

I would also add that `raw_exec` uses the same user permissions as the Nomad process. So if Nomad runs as root / administrator the forked process also runs as that user.

**Comment by cbednarski** [comment]
A few minor comments, but LGTM!

File: client/driver/raw_exec.go (hunk 1 above)
Comment (dadgar) [inline client/driver/raw_exec.go]: Nice I didn't know about this :+1:

File: client/driver/raw_exec.go (hunk 2 above)
Comment (dadgar) [inline client/driver/raw_exec.go]: Tasks are only added in that map after they have been made. So we are safe here

**Comment by jippi** [comment]
nice - is there an ETA for a binary release with this included?

**Comment by cbednarski** [comment]
@jippi This will come out in 0.2.0 but we don't have a release date yet. If you'd like to try this out in advance you can build from master.

**Pull request merged**